###### 1. import the ODI_cricker.xlsx file bowler sheet in your notebook

In [95]:
import pandas as pd

In [96]:
df = pd.read_excel("ODI_cricket.xlsx", sheet_name='bowler', engine="openpyxl")

###### 2. Display the first 10 rows of the dataframe


In [97]:
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

###### 3. Create a markdown cell and explain the meaning of each column

In [98]:
print(df.columns)

Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'Ave',   'Econ',     'SR',        4,        5],
      dtype='object')


###### Player- Player name and county/team
<br>Span- Playing span
<br>Mat- Matches played
<br>Inns- Innings bowled
<br>Balls- Balls bowled
<br>Runs-Runs conceded
<br>Wkts- Wickets taken
<br>Ave- Bowling average
<br>Econ- Economy rate
<br>SR- Bowling strike rate
<br>4- four wkts in an inns
<br>5- five wkts in an inns

###### 4. find the number of rows and columns in the dataframe

In [99]:
print('number of rows=',df.shape[0])
print('number of rows=',df.shape[1])

number of rows= 77
number of rows= 12


###### 5. find the data statistics and check for the data types

In [100]:
display(df.describe())
display(df.info())

Mat        Inns         Balls          Runs        Wkts  \
count   77.000000   77.000000     77.000000     77.000000   77.000000   
mean   194.402597  181.194805   8839.402597   6671.714286  233.805195   
std     82.485606   67.958393   3316.055457   2245.839029   84.406603   
min     80.000000   76.000000   4074.000000   2821.000000  151.000000   
25%    136.000000  128.000000   6182.000000   5058.000000  173.000000   
50%    170.000000  164.000000   8054.000000   6192.000000  199.000000   
75%    227.000000  218.000000  10750.000000   8021.000000  272.000000   
max    463.000000  372.000000  18811.000000  13632.000000  534.000000   

             Ave       Econ         SR          4          5  
count  77.000000  77.000000  77.000000  77.000000  77.000000  
mean   28.958052   4.596753  37.909091   6.350649   2.870130  
std     4.826768   0.515814   6.060901   3.556929   2.530606  
min    18.680000   3.300000  26.100000   1.000000   0.000000  
25%    24.970000   4.280000  33.000000   3.000000   1.000000  
50%    29.290000   4.660000  37.800000   6.000000   2.000000  
75%    31.900000   4.920000  41.400000   8.000000   4.000000  
max    44.480000   5.830000  52.500000  17.000000  13.000000

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  77 non-null     object 
 1   Span    77 non-null     object 
 2   Mat     77 non-null     int64  
 3   Inns    77 non-null     int64  
 4   Balls   77 non-null     int64  
 5   Runs    77 non-null     int64  
 6   Wkts    77 non-null     int64  
 7   Ave     77 non-null     float64
 8   Econ    77 non-null     float64
 9   SR      77 non-null     float64
 10  4       77 non-null     int64  
 11  5       77 non-null     int64  
dtypes: float64(3), int64(7), object(2)
memory usage: 7.3+ KB


None

###### 6. Are there any missing values present in the dataset?

In [101]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  77 non-null     object 
 1   Span    77 non-null     object 
 2   Mat     77 non-null     int64  
 3   Inns    77 non-null     int64  
 4   Balls   77 non-null     int64  
 5   Runs    77 non-null     int64  
 6   Wkts    77 non-null     int64  
 7   Ave     77 non-null     float64
 8   Econ    77 non-null     float64
 9   SR      77 non-null     float64
 10  4       77 non-null     int64  
 11  5       77 non-null     int64  
dtypes: float64(3), int64(7), object(2)
memory usage: 7.3+ KB


None

###### From above info, as there is no null value, we can say there is no missing value.

###### 7. Renaming the column names appropriately

In [102]:
print(df.columns)

Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'Ave',   'Econ',     'SR',        4,        5],
      dtype='object')


In [103]:
df = df.rename(columns={'Span':'Playing_span',
                        'Mat':'Matches_played',
                        'Inns':'Innings_bowled',
                        'Balls':'Balls_bowled',
                        'Runs':'Runs_conceded',
                        'Wkts':'Wickets_taken',
                        'Ave':'Bowling_average',
                        'Econ':'Economy_rate',
                        'SR':'Bowling_strike_rate',
                        4:'Four_wkts_in_an_inns',
                        5:'Five_wkts_in_an_inns'})

In [104]:
df.head()

Player Playing_span  Matches_played  Innings_bowled  \
0  M Muralitharan (Asia/ICC/SL)    1993-2011             350             341   
1             Wasim Akram (PAK)    1984-2003             356             351   
2            Waqar Younis (PAK)    1989-2003             262             258   
3          WPUJC Vaas (Asia/SL)    1994-2008             322             320   
4  Shahid Afridi (Asia/ICC/PAK)    1996-2015             398             372   

   Balls_bowled  Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0         18811          12326            534            23.08          3.93   
1         18186          11812            502            23.52          3.89   
2         12698           9919            416            23.84          4.68   
3         15775          11014            400            27.53          4.18   
4         17670          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  
0                 35.2                    15                    10  
1                 36.2                    17                     6  
2                 30.5                    14                    13  
3                 39.4                     9                     4  
4                 44.7                     4                     9

###### 8. How many players played for ICC?

In [105]:
def icc_check(x):
    if 'ICC' in x:
        return 'Yes'
    else:
        return 'No'

In [106]:
df['played_for_ICC'] = df['Player'].apply(icc_check)

In [107]:
display(df.head())

Player Playing_span  Matches_played  Innings_bowled  \
0  M Muralitharan (Asia/ICC/SL)    1993-2011             350             341   
1             Wasim Akram (PAK)    1984-2003             356             351   
2            Waqar Younis (PAK)    1989-2003             262             258   
3          WPUJC Vaas (Asia/SL)    1994-2008             322             320   
4  Shahid Afridi (Asia/ICC/PAK)    1996-2015             398             372   

   Balls_bowled  Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0         18811          12326            534            23.08          3.93   
1         18186          11812            502            23.52          3.89   
2         12698           9919            416            23.84          4.68   
3         15775          11014            400            27.53          4.18   
4         17670          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC  
0            Yes  
1             No  
2             No  
3             No  
4            Yes

In [108]:
print(df['played_for_ICC'].value_counts())

No     64
Yes    13
Name: played_for_ICC, dtype: int64


###### 13 players played for ICC.

###### 9. How many different countries are present in this dataset? 

In [109]:
df[['Player_Name', 'Country']] = df['Player'].str.split('(', expand=True)

In [110]:
display(df.head())

Player Playing_span  Matches_played  Innings_bowled  \
0  M Muralitharan (Asia/ICC/SL)    1993-2011             350             341   
1             Wasim Akram (PAK)    1984-2003             356             351   
2            Waqar Younis (PAK)    1989-2003             262             258   
3          WPUJC Vaas (Asia/SL)    1994-2008             322             320   
4  Shahid Afridi (Asia/ICC/PAK)    1996-2015             398             372   

   Balls_bowled  Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0         18811          12326            534            23.08          3.93   
1         18186          11812            502            23.52          3.89   
2         12698           9919            416            23.84          4.68   
3         15775          11014            400            27.53          4.18   
4         17670          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC      Player_Name        Country  
0            Yes  M Muralitharan    Asia/ICC/SL)  
1             No     Wasim Akram            PAK)  
2             No    Waqar Younis            PAK)  
3             No      WPUJC Vaas        Asia/SL)  
4            Yes   Shahid Afridi   Asia/ICC/PAK)

In [111]:
df.drop('Player', axis=1, inplace=True)

In [112]:
display(df.head())

Playing_span  Matches_played  Innings_bowled  Balls_bowled  Runs_conceded  \
0    1993-2011             350             341         18811          12326   
1    1984-2003             356             351         18186          11812   
2    1989-2003             262             258         12698           9919   
3    1994-2008             322             320         15775          11014   
4    1996-2015             398             372         17670          13632   

   Wickets_taken  Bowling_average  Economy_rate  Bowling_strike_rate  \
0            534            23.08          3.93                 35.2   
1            502            23.52          3.89                 36.2   
2            416            23.84          4.68                 30.5   
3            400            27.53          4.18                 39.4   
4            395            34.51          4.62                 44.7   

   Four_wkts_in_an_inns  Five_wkts_in_an_inns played_for_ICC      Player_Name  \
0                    15                    10            Yes  M Muralitharan    
1                    17                     6             No     Wasim Akram    
2                    14                    13             No    Waqar Younis    
3                     9                     4             No      WPUJC Vaas    
4                     4                     9            Yes   Shahid Afridi    

         Country  
0   Asia/ICC/SL)  
1           PAK)  
2           PAK)  
3       Asia/SL)  
4  Asia/ICC/PAK)

In [113]:
df['Country'] = df['Country'].str.replace(")", "")

C:\Users\faisa\AppData\Local\Temp/ipykernel_6096/381900187.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Country'] = df['Country'].str.replace(")", "")


In [114]:
display(df.head())

Playing_span  Matches_played  Innings_bowled  Balls_bowled  Runs_conceded  \
0    1993-2011             350             341         18811          12326   
1    1984-2003             356             351         18186          11812   
2    1989-2003             262             258         12698           9919   
3    1994-2008             322             320         15775          11014   
4    1996-2015             398             372         17670          13632   

   Wickets_taken  Bowling_average  Economy_rate  Bowling_strike_rate  \
0            534            23.08          3.93                 35.2   
1            502            23.52          3.89                 36.2   
2            416            23.84          4.68                 30.5   
3            400            27.53          4.18                 39.4   
4            395            34.51          4.62                 44.7   

   Four_wkts_in_an_inns  Five_wkts_in_an_inns played_for_ICC      Player_Name  \
0                    15                    10            Yes  M Muralitharan    
1                    17                     6             No     Wasim Akram    
2                    14                    13             No    Waqar Younis    
3                     9                     4             No      WPUJC Vaas    
4                     4                     9            Yes   Shahid Afridi    

        Country  
0   Asia/ICC/SL  
1           PAK  
2           PAK  
3       Asia/SL  
4  Asia/ICC/PAK

In [115]:
print(df['Country'].value_counts())

INDIA           9
PAK             9
AUS             8
NZ              6
SL              5
WI              5
Asia/INDIA      4
ENG             4
Asia/SL         3
SA              3
Afr/ICC/SA      2
AUS/ICC         2
Asia/ICC/PAK    2
ENG/ICC         2
ICC/NZ          2
BAN             2
Afr/SA          2
ICC/WI          1
Asia/ICC/SL     1
Afr/ZIM         1
ICC/SA          1
Asia/PAK        1
Asia/BAN        1
AFG             1
Name: Country, dtype: int64


In [116]:
def Asia_check(x):
    if 'Asia' in x:
        return 'Yes'
    else:
        return 'No'

In [117]:
def Afr_check(x):
    if 'Afr' in x:
        return 'Yes'
    else:
        return 'No'

In [118]:
df['played_for_Asia'] = df['Country'].apply(Asia_check)
df['played_for_Afr'] = df['Country'].apply(Afr_check)

In [119]:
display(df.head())

Playing_span  Matches_played  Innings_bowled  Balls_bowled  Runs_conceded  \
0    1993-2011             350             341         18811          12326   
1    1984-2003             356             351         18186          11812   
2    1989-2003             262             258         12698           9919   
3    1994-2008             322             320         15775          11014   
4    1996-2015             398             372         17670          13632   

   Wickets_taken  Bowling_average  Economy_rate  Bowling_strike_rate  \
0            534            23.08          3.93                 35.2   
1            502            23.52          3.89                 36.2   
2            416            23.84          4.68                 30.5   
3            400            27.53          4.18                 39.4   
4            395            34.51          4.62                 44.7   

   Four_wkts_in_an_inns  Five_wkts_in_an_inns played_for_ICC      Player_Name  \
0                    15                    10            Yes  M Muralitharan    
1                    17                     6             No     Wasim Akram    
2                    14                    13             No    Waqar Younis    
3                     9                     4             No      WPUJC Vaas    
4                     4                     9            Yes   Shahid Afridi    

        Country played_for_Asia played_for_Afr  
0   Asia/ICC/SL             Yes             No  
1           PAK              No             No  
2           PAK              No             No  
3       Asia/SL             Yes             No  
4  Asia/ICC/PAK             Yes             No

In [120]:
display(df.columns)

Index(['Playing_span', 'Matches_played', 'Innings_bowled', 'Balls_bowled',
       'Runs_conceded', 'Wickets_taken', 'Bowling_average', 'Economy_rate',
       'Bowling_strike_rate', 'Four_wkts_in_an_inns', 'Five_wkts_in_an_inns',
       'played_for_ICC', 'Player_Name', 'Country', 'played_for_Asia',
       'played_for_Afr'],
      dtype='object')

In [121]:
new_col_sequence = ['Player_Name', 'Country', 'Playing_span', 'Matches_played', 'Innings_bowled', 'Balls_bowled',
       'Runs_conceded', 'Wickets_taken', 'Bowling_average', 'Economy_rate',
       'Bowling_strike_rate', 'Four_wkts_in_an_inns', 'Five_wkts_in_an_inns',
       'played_for_ICC', 'played_for_Asia',
       'played_for_Afr']

In [122]:
display(df.head())

Playing_span  Matches_played  Innings_bowled  Balls_bowled  Runs_conceded  \
0    1993-2011             350             341         18811          12326   
1    1984-2003             356             351         18186          11812   
2    1989-2003             262             258         12698           9919   
3    1994-2008             322             320         15775          11014   
4    1996-2015             398             372         17670          13632   

   Wickets_taken  Bowling_average  Economy_rate  Bowling_strike_rate  \
0            534            23.08          3.93                 35.2   
1            502            23.52          3.89                 36.2   
2            416            23.84          4.68                 30.5   
3            400            27.53          4.18                 39.4   
4            395            34.51          4.62                 44.7   

   Four_wkts_in_an_inns  Five_wkts_in_an_inns played_for_ICC      Player_Name  \
0                    15                    10            Yes  M Muralitharan    
1                    17                     6             No     Wasim Akram    
2                    14                    13             No    Waqar Younis    
3                     9                     4             No      WPUJC Vaas    
4                     4                     9            Yes   Shahid Afridi    

        Country played_for_Asia played_for_Afr  
0   Asia/ICC/SL             Yes             No  
1           PAK              No             No  
2           PAK              No             No  
3       Asia/SL             Yes             No  
4  Asia/ICC/PAK             Yes             No

In [123]:
df = df[new_col_sequence]

In [124]:
display(df.head())

Player_Name       Country Playing_span  Matches_played  Innings_bowled  \
0  M Muralitharan    Asia/ICC/SL    1993-2011             350             341   
1     Wasim Akram            PAK    1984-2003             356             351   
2    Waqar Younis            PAK    1989-2003             262             258   
3      WPUJC Vaas        Asia/SL    1994-2008             322             320   
4   Shahid Afridi   Asia/ICC/PAK    1996-2015             398             372   

   Balls_bowled  Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0         18811          12326            534            23.08          3.93   
1         18186          11812            502            23.52          3.89   
2         12698           9919            416            23.84          4.68   
3         15775          11014            400            27.53          4.18   
4         17670          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr  
0            Yes             Yes             No  
1             No              No             No  
2             No              No             No  
3             No             Yes             No  
4            Yes             Yes             No

In [125]:
print(df['Country'].value_counts())

INDIA           9
PAK             9
AUS             8
NZ              6
SL              5
WI              5
Asia/INDIA      4
ENG             4
Asia/SL         3
SA              3
Afr/ICC/SA      2
AUS/ICC         2
Asia/ICC/PAK    2
ENG/ICC         2
ICC/NZ          2
BAN             2
Afr/SA          2
ICC/WI          1
Asia/ICC/SL     1
Afr/ZIM         1
ICC/SA          1
Asia/PAK        1
Asia/BAN        1
AFG             1
Name: Country, dtype: int64


In [126]:
df['Country'] = df['Country'].str.replace('Asia','')
df['Country'] = df['Country'].str.replace('ICC','')
df['Country'] = df['Country'].str.replace('Afr','')
df['Country'] = df['Country'].str.replace('/','')

In [127]:
df.nunique()

Player_Name             77
Country                 11
Playing_span            66
Matches_played          67
Innings_bowled          68
Balls_bowled            76
Runs_conceded           76
Wickets_taken           59
Bowling_average         72
Economy_rate            60
Bowling_strike_rate     63
Four_wkts_in_an_inns    15
Five_wkts_in_an_inns    12
played_for_ICC           2
played_for_Asia          2
played_for_Afr           2
dtype: int64

###### From the above list, there are 11 different countries are present in this dataset

###### 10. Which player(s) had played for the longest period of time?

In [128]:
df['Start_year'] = df['Playing_span'].str[:4]
df['End_year'] = df['Playing_span'].str[5:]

In [129]:
display(df.head())

Player_Name Country Playing_span  Matches_played  Innings_bowled  \
0  M Muralitharan       SL    1993-2011             350             341   
1     Wasim Akram      PAK    1984-2003             356             351   
2    Waqar Younis      PAK    1989-2003             262             258   
3      WPUJC Vaas       SL    1994-2008             322             320   
4   Shahid Afridi      PAK    1996-2015             398             372   

   Balls_bowled  Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0         18811          12326            534            23.08          3.93   
1         18186          11812            502            23.52          3.89   
2         12698           9919            416            23.84          4.68   
3         15775          11014            400            27.53          4.18   
4         17670          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr Start_year End_year  
0            Yes             Yes             No       1993     2011  
1             No              No             No       1984     2003  
2             No              No             No       1989     2003  
3             No             Yes             No       1994     2008  
4            Yes             Yes             No       1996     2015

In [130]:
df=df.drop('Playing_span', axis=1)

In [131]:
df.head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
0  M Muralitharan       SL             350             341         18811   
1     Wasim Akram      PAK             356             351         18186   
2    Waqar Younis      PAK             262             258         12698   
3      WPUJC Vaas       SL             322             320         15775   
4   Shahid Afridi      PAK             398             372         17670   

   Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0          12326            534            23.08          3.93   
1          11812            502            23.52          3.89   
2           9919            416            23.84          4.68   
3          11014            400            27.53          4.18   
4          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr Start_year End_year  
0            Yes             Yes             No       1993     2011  
1             No              No             No       1984     2003  
2             No              No             No       1989     2003  
3             No             Yes             No       1994     2008  
4            Yes             Yes             No       1996     2015

In [132]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Player_Name           77 non-null     object 
 1   Country               77 non-null     object 
 2   Matches_played        77 non-null     int64  
 3   Innings_bowled        77 non-null     int64  
 4   Balls_bowled          77 non-null     int64  
 5   Runs_conceded         77 non-null     int64  
 6   Wickets_taken         77 non-null     int64  
 7   Bowling_average       77 non-null     float64
 8   Economy_rate          77 non-null     float64
 9   Bowling_strike_rate   77 non-null     float64
 10  Four_wkts_in_an_inns  77 non-null     int64  
 11  Five_wkts_in_an_inns  77 non-null     int64  
 12  played_for_ICC        77 non-null     object 
 13  played_for_Asia       77 non-null     object 
 14  played_for_Afr        77 non-null     object 
 15  Start_year            77 

In [133]:
df['Start_year'] = df['Start_year'].astype('int') 
df['End_year'] = df['End_year'].astype('int')

In [134]:
df.head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
0  M Muralitharan       SL             350             341         18811   
1     Wasim Akram      PAK             356             351         18186   
2    Waqar Younis      PAK             262             258         12698   
3      WPUJC Vaas       SL             322             320         15775   
4   Shahid Afridi      PAK             398             372         17670   

   Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0          12326            534            23.08          3.93   
1          11812            502            23.52          3.89   
2           9919            416            23.84          4.68   
3          11014            400            27.53          4.18   
4          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr  Start_year  End_year  
0            Yes             Yes             No        1993      2011  
1             No              No             No        1984      2003  
2             No              No             No        1989      2003  
3             No             Yes             No        1994      2008  
4            Yes             Yes             No        1996      2015

In [135]:
df['Years_played'] = df['End_year'] - df['Start_year']+1

In [136]:
df.head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
0  M Muralitharan       SL             350             341         18811   
1     Wasim Akram      PAK             356             351         18186   
2    Waqar Younis      PAK             262             258         12698   
3      WPUJC Vaas       SL             322             320         15775   
4   Shahid Afridi      PAK             398             372         17670   

   Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0          12326            534            23.08          3.93   
1          11812            502            23.52          3.89   
2           9919            416            23.84          4.68   
3          11014            400            27.53          4.18   
4          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr  Start_year  End_year  \
0            Yes             Yes             No        1993      2011   
1             No              No             No        1984      2003   
2             No              No             No        1989      2003   
3             No             Yes             No        1994      2008   
4            Yes             Yes             No        1996      2015   

   Years_played  
0            19  
1            20  
2            15  
3            15  
4            20

In [137]:
df = df.drop(['Start_year','End_year'], axis=1)

In [138]:
df.head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
0  M Muralitharan       SL             350             341         18811   
1     Wasim Akram      PAK             356             351         18186   
2    Waqar Younis      PAK             262             258         12698   
3      WPUJC Vaas       SL             322             320         15775   
4   Shahid Afridi      PAK             398             372         17670   

   Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
0          12326            534            23.08          3.93   
1          11812            502            23.52          3.89   
2           9919            416            23.84          4.68   
3          11014            400            27.53          4.18   
4          13632            395            34.51          4.62   

   Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
0                 35.2                    15                    10   
1                 36.2                    17                     6   
2                 30.5                    14                    13   
3                 39.4                     9                     4   
4                 44.7                     4                     9   

  played_for_ICC played_for_Asia played_for_Afr  Years_played  
0            Yes             Yes             No            19  
1             No              No             No            20  
2             No              No             No            15  
3             No             Yes             No            15  
4            Yes             Yes             No            20

In [139]:
df.sort_values(by='Years_played', ascending = False).head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
72   SR Tendulkar    INDIA             463             270          8054   
10  ST Jayasuriya       SL             445             368         14874   
66   Shoaib Malik      PAK             287             217          7958   
62       CH Gayle       WI             301             199          7424   
4   Shahid Afridi      PAK             398             372         17670   

    Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
72           6850            154            44.48          5.10   
10          11871            323            36.75          4.78   
66           6192            158            39.18          4.66   
62           5926            167            35.48          4.78   
4           13632            395            34.51          4.62   

    Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
72                 52.2                     4                     2   
10                 46.0                     8                     4   
66                 50.3                     1                     0   
62                 44.4                     3                     1   
4                  44.7                     4                     9   

   played_for_ICC played_for_Asia played_for_Afr  Years_played  
72             No              No             No            24  
10             No             Yes             No            23  
66             No              No             No            21  
62            Yes              No             No            21  
4             Yes             Yes             No            20

###### From the sorted list above SR Tendulkar had played for the longest period of time, 24 years.

###### 11. Which player(s) had played for the shortest period of time?

In [140]:
df.sort_values(by='Years_played').head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
40   BKV Prasad    INDIA             161             160          8129   
74  Rashid Khan      AFG              80              76          4074   
73   BAW Mendis       SL              87              84          4154   
49  Saeed Ajmal      PAK             113             112          6000   
44   L Klusener       SA             171             164          7336   

    Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
40           6332            196            32.30          4.67   
74           2821            151            18.68          4.15   
73           3324            152            21.86          4.80   
49           4182            184            22.72          4.18   
44           5751            192            29.95          4.70   

    Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
40                 41.4                     3                     1   
74                 26.9                     5                     4   
73                 27.3                     7                     3   
49                 32.6                     6                     2   
44                 38.2                     1                     6   

   played_for_ICC played_for_Asia played_for_Afr  Years_played  
40             No              No             No             8  
74             No              No             No             8  
73             No              No             No             8  
49             No              No             No             8  
44             No              No             No             9

###### From the sorted list above BKV Prasad	had played for the shortest period of time, 8 years.

###### 12. How many Australian Bowlers are present in this dataset?

In [141]:
print(df['Country'].value_counts())

INDIA    13
PAK      12
AUS      10
SL        9
SA        8
NZ        8
ENG       6
WI        6
BAN       3
ZIM       1
AFG       1
Name: Country, dtype: int64


###### 10 Australian Bowlers are present in this dataset.

###### 13. Is there any Bangladeshi player present in this dataset?
<br>Yes, 3 Bangladeshi players are present in this dataset.

###### 14. Which player had the lowest economy rate?

In [142]:
df.sort_values(by='Economy_rate').head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
65    RJ Hadlee       NZ             115             112          6182   
32  CEL Ambrose       WI             176             175          9353   
67  MD Marshall       WI             136             134          7175   
5    SM Pollock       SA             303             297         15712   
25  N Kapil Dev    INDIA             225             221         11202   

    Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
65           3407            158            21.56          3.30   
32           5429            225            24.12          3.48   
67           4233            157            26.96          3.53   
5            9631            393            24.50          3.67   
25           6945            253            27.45          3.71   

    Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
65                 39.1                     1                     5   
32                 41.5                     6                     4   
67                 45.7                     6                     0   
5                  39.9                    12                     5   
25                 44.2                     3                     1   

   played_for_ICC played_for_Asia played_for_Afr  Years_played  
65             No              No             No            18  
32             No              No             No            13  
67             No              No             No            13  
5             Yes              No            Yes            13  
25             No              No             No            17

###### RJ Hadlee had the lowest economy rate, 3.30.

###### 15. Which player had the lowest strike rate?

In [143]:
df.sort_values(by='Bowling_strike_rate').head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
41     MA Starc      AUS              99              99          5099   
74  Rashid Khan      AFG              80              76          4074   
73   BAW Mendis       SL              87              84          4154   
7         B Lee      AUS             221             217         11185   
60     TA Boult       NZ              93              93          5117   

    Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
41           4379            195            22.45          5.15   
74           2821            151            18.68          4.15   
73           3324            152            21.86          4.80   
7            8877            380            23.36          4.76   
60           4261            169            25.21          4.99   

    Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
41                 26.1                    11                     8   
74                 26.9                     5                     4   
73                 27.3                     7                     3   
7                  29.4                    14                     9   
60                 30.2                     8                     5   

   played_for_ICC played_for_Asia played_for_Afr  Years_played  
41             No              No             No            12  
74             No              No             No             8  
73             No              No             No             8  
7              No              No             No            13  
60             No              No             No            10

###### MA Starc had the lowest strike rate, 26.1.

###### 16. Which player had the lowest bowling average?

In [144]:
df.sort_values(by='Bowling_average').head()

Player_Name Country  Matches_played  Innings_bowled  Balls_bowled  \
74      Rashid Khan      AFG              80              76          4074   
65        RJ Hadlee       NZ             115             112          6182   
19        AA Donald       SA             164             162          8561   
14  Saqlain Mushtaq      PAK             169             165          8770   
73       BAW Mendis       SL              87              84          4154   

    Runs_conceded  Wickets_taken  Bowling_average  Economy_rate  \
74           2821            151            18.68          4.15   
65           3407            158            21.56          3.30   
19           5926            272            21.78          4.15   
14           6275            288            21.78          4.29   
73           3324            152            21.86          4.80   

    Bowling_strike_rate  Four_wkts_in_an_inns  Five_wkts_in_an_inns  \
74                 26.9                     5                     4   
65                 39.1                     1                     5   
19                 31.4                    11                     2   
14                 30.4                    11                     6   
73                 27.3                     7                     3   

   played_for_ICC played_for_Asia played_for_Afr  Years_played  
74             No              No             No             8  
65             No              No             No            18  
19             No              No             No            13  
14             No              No             No             9  
73             No              No             No             8

###### Rashid Khan had the lowest bowling average, 18.68.

###### 17. Remove Unnecessary columns if needed

In [145]:
display(df.columns)

Index(['Player_Name', 'Country', 'Matches_played', 'Innings_bowled',
       'Balls_bowled', 'Runs_conceded', 'Wickets_taken', 'Bowling_average',
       'Economy_rate', 'Bowling_strike_rate', 'Four_wkts_in_an_inns',
       'Five_wkts_in_an_inns', 'played_for_ICC', 'played_for_Asia',
       'played_for_Afr', 'Years_played'],
      dtype='object')

###### Unnecessary columns were removed during the operations, so which columns are left, all seems to be necessary.